### ISSUE TITLE
Closure conversion: for loops

### ISSUE THREAD

**Issue by peter-ahe-google** [dart-lang/kernel#32]
Closure conversion needs to support capturing the for-loop variable and pay special attention to the fact that a new variable is introduced per iteration.

**Comment by peter-ahe-google** [comment]
Proposed fix in [CL 520497013](https://chromereviews.googleplex.com/520497013).

**Comment by peter-ahe-google** [comment]
Fixed by 34036c5aac323fe251db558ab9986eaa06408602.

**Comment by peter-ahe-google** [comment]
The fix wasn't correct.

@asgerf has made a [test](https://github.com/dart-lang/sdk/blob/master/tests/language/for_variable_capture_test.dart) that demonstrates why.

The problem is that other initializers can capture previously defined loop variables, and that updates can also capture loop variables. For example:

```dart
  var closure;
  for (var i=0, fn = () => i; i < 3; i++) {
    i += 1;
    closure = fn;
  }
  Expect.equals(1, closure());
```

**Comment by peter-ahe-google** [comment]
Fixed by 6b464934860f5898ea71285306096ff60bad4563.
